src/controllers/state_controller.hpp

```diff
@@ -858,6 +951,42 @@ void AerState::apply_mcphase(const reg_t &qubits, const std::complex<double> pha
   buffer_op(std::move(op));
 }
 
+void AerState::apply_h(const uint_t qubit) {
+  assert_initialized();
+
+  Operations::Op op;
+  op.type = Operations::OpType::gate;
+  op.name = "h";
+  op.qubits = {qubit};
+  op.params = {};
+
+  buffer_op(std::move(op));
+}
+
+void AerState::apply_u(const uint_t qubit, const double theta, const double phi, const double lambda) {
+  assert_initialized();
+
+  Operations::Op op;
+  op.type = Operations::OpType::gate;
+  op.name = "u";
+  op.qubits = {qubit};
+  op.params = {theta, phi, lambda, 0.0};
+
+  buffer_op(std::move(op));
+}
+
+void AerState::apply_cu(const reg_t &qubits, const double theta, const double phi, const double lambda) {
+  assert_initialized();
+
+  Operations::Op op;
+  op.type = Operations::OpType::gate;
+  op.name = "cu";
+  op.qubits = qubits;
+  op.params = {theta, phi, lambda, 0.0};
+
+  buffer_op(std::move(op));
```

Comment:
This doesn't seem to have an analogue in the Python components - is it meant to?  Also, Terra's `cu` takes 4 parameters like Aer, so it could accept the last one in the signature as well.

Reply:
Oh wait whoops sorry - it _does_ have a Python-space component, and I even commented on it.  The 4-parameter thing is still a question, though.

Reply:
https://github.com/Qiskit/qiskit-aer/blob/cd8710f3ec83391dbbe7f67d0c185c2b0538b17f/src/simulators/statevector/statevector_state.hpp#L264-L269

Aer supports U4.

Reply:
Sorry, the thing I mean is that Terra's `cu` is a 4-parameter gate.  This definition in Aer only takes three parameters, so there seems to be a mismatch of expectations between them.